gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcProvider.java

```diff
@@ -21,18 +27,50 @@
  * @author nveeramr
  */
 public class JdbcProvider extends BasicDataSource {
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcProvider.class);
+  private Tunnel tunnel;
+
   // If extract type is not provided then consider it as a default type
   public JdbcProvider(String driver, String connectionUrl, String user, String password, int numconn, int timeout) {
-    this.connect(driver, connectionUrl, user, password, numconn, timeout, "DEFAULT");
+    this.connect(driver, connectionUrl, user, password, numconn, timeout, "DEFAULT", null, -1);
   }
 
   public JdbcProvider(String driver, String connectionUrl, String user, String password, int numconn, int timeout,
       String type) {
-    this.connect(driver, connectionUrl, user, password, numconn, timeout, type);
+    this.connect(driver, connectionUrl, user, password, numconn, timeout, type, null, -1);
+  }
+
+  public JdbcProvider(String driver, String connectionUrl, String user, String password, int numconn, int timeout,
+      String type, String proxyHost, int proxyPort) {
+    this.connect(driver, connectionUrl, user, password, numconn, timeout, type, proxyHost, proxyPort);
   }
 
   public void connect(String driver, String connectionUrl, String user, String password, int numconn, int timeout,
-      String type) {
+      String type, String proxyHost, int proxyPort) {
+
+    if (proxyHost != null && proxyPort > 0) {
+      String remoteHost = "";
+      int remotePort = 0;
+      // TODO make connection Url parsing much more robust -- some connections URLs can have colons and slashes in the
+      // weirdest places
+      int hostStart = connectionUrl.indexOf("://") + 3;
+      int portStart =  connectionUrl.indexOf(":", hostStart);
+      remoteHost = connectionUrl.substring(hostStart, portStart);
+      remotePort = Integer.decode(connectionUrl.substring(portStart + 1, connectionUrl.indexOf("/", portStart)));
```

Comment:
I think you can simply the `connectionUrl` parsing by creating a (URI)[https://docs.oracle.com/javase/7/docs/api/java/net/URI.html] via `URI.create(connectionUrl)` and then using methods such as `getScheme()` or `getPort()` to retrieve the relevant information.